legacy-contracts/locked-asset-distribution/src/lib.rs

```diff
@@ -42,15 +42,20 @@ pub trait Distribution {
     }
 
     #[endpoint(clearCommunityDistributionList)]
-    fn clear_community_distribution_list(&self, n: u64) -> usize {
-        for _ in 0..n {
-            self.community_distribution_list().pop_front();
+    fn clear_community_distribution_list(&self, n: u64) -> (u64, usize) {
+        let mut counter = 0;
+        for node in self.community_distribution_list().iter() {
+            self.community_distribution_list().remove_node(&node);
+            counter += 1;
+            if counter >= n {
+                break;
+            }
         }
-        self.community_distribution_list().len()
+        (counter, self.community_distribution_list().len())
     }
 
     #[endpoint(clearUserLockedAssetMap)]
-    fn clear_user_locked_asset_map(&self, n: u64) -> usize {
+    fn clear_user_locked_asset_map(&self, n: u64) -> (u64, usize) {
```

Comment:
Also, maybe add a comment about what is returned, like
// Returns number of processed elements and remaining elements in the storage.